tools/ci/release-comment.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+"""
+python3 release-comment.py <repo owner> <repo name> <release tag> <PR snippets ...>
+
+This script takes paths to some number of changelog snippets with filenames in
+the form "pr-PRNUM.md" and comments on each PR and associated issue(s)
+announcing the release of the PR in the GitHub release with the given tag.
+
+Requirements:
+- Python 3.7+
+- requests
+"""
+
+from __future__ import annotations
+
+import argparse
+import json
+import os
+import os.path
+import re
+import sys
+from dataclasses import (
+    InitVar,
+    dataclass,
+    field,
+)
+
+import requests
+
+GITHUB_API_URL = os.environ.get("GITHUB_API_URL", "https://api.github.com")
+
+GRAPHQL_API_URL = os.environ.get("GITHUB_GRAPHQL_URL", "https://api.github.com/graphql")
+
+
+@dataclass
+class ReleaseCommenter:
+    repo_owner: str
+    repo_name: str
+    release_tag: str
+    token: InitVar[str]
+    session: requests.Session = field(init=False)
+
+    def __post_init__(self, token: str) -> None:
+        self.session = requests.Session()
+        self.session.headers["Authorization"] = f"bearer {token}"
+
+    def __enter__(self) -> ReleaseCommenter:
+        return self
+
+    def __exit__(self, *_exc) -> None:
+        self.session.close()
+
+    @property
+    def release_url(self) -> str:
+        return f"https://github.com/{self.repo_owner}/{self.repo_name}/releases/tag/{self.release_tag}"
```

Comment:
code looks right to me here but the only demo issue at https://github.com/jwodder/pr-merge-release-test/issues/8#issuecomment-1245613726 seems to have wrong url to release: [0.0.4](https://github.com/%7Bself.repo_owner%7D/%7Bself.repo_name%7D/releases/tag/%7Bself.release_tag%7D) -- may be was fixed up after that issue closed/released?

Reply:
Yes, it was fixed in https://github.com/jwodder/pr-merge-release-test/pull/10 (and the comment on that one has a valid release URL).

Reply:
cool, thanks!